integration/tests/cook/test_multi_user.py

```diff
@@ -252,6 +253,64 @@ def test_rate_limit_while_creating_job(self):
             finally:
                 util.kill_jobs(self.cook_url,jobs_to_kill)
 
+    def test_rate_limit_launching_jobs(self):
+        settings = util.settings(self.cook_url)
+        if settings['rate-limit']['job-launch'] is None:
+            pytest.skip()
+
+        # Allow an environmental variable override.
+        user = None
+        name = os.getenv('COOK_LAUNCH_RATE_LIMIT_NAME')
+        if name is not None:
+            user = self.user_factory.user_class(name)
+        else:
+            user = self.user_factory.new_user()
+
+        bucket_size = settings['rate-limit']['job-launch']['bucket-size']
+        token_rate = settings['rate-limit']['job-launch']['tokens-replenished-per-minute']
+        if token_rate < 5 or token_rate > 20:
+            pytest.skip()
+        if bucket_size < 10 or bucket_size > 20:
+            pytest.skip()
+        with user:
+            job_uuids = []
+            try:
+                jobspec = {"command": "sleep 240", 'cpus': 0.03, 'mem': 32}
```

Comment:
0.03 seems really specific. Is that needed?

Reply:
It needs to be something small so that we don't hit the mesos CPU quota and fail to launch for that reason, not because of quota. The acceptable range depends on the token bucket filter parameters + some unknown fudge factors, cause travis.